rules/engine.go

```diff
@@ -166,11 +167,18 @@ func (e *v3Engine) SetWatcherWrapper(watcherWrapper WrapWatcher) {
 	e.watcherWrapper = watcherWrapper
 }
 
+// valid path patterns must be alphanumeric and may only contain select special characters (:/"'_.,*=-)
+var validPath = regexp.MustCompile(`^[[:alnum:] \:\/\"\'\_\.\,\*\=\-]*$`)
+
 func (e *v3Engine) AddRule(rule DynamicRule,
 	lockPattern string,
 	callback V3RuleTaskCallback,
 	options ...RuleOption) {
-	e.addRuleWithIface(rule, lockPattern, callback, options...)
+	if !validPath.MatchString(lockPattern) {
```

Comment:
Not including check that the path contains "lock" since `/crawler/compliance-engine
/armada-ingress/:region/clusters/:clusterid/ingress_update` paths don't contain

Reply:
Humm, I think we should push to get that changed and then announce in armada-dev that this will be required from now on.